common/app/routes/challenges/redux/selectors.js

```diff
@@ -36,14 +36,26 @@ const submitTypes = {
 export const challengeSelector = createSelector(
   state => state.challengesApp.challenge,
   state => state.entities.challenge,
-  (challengeName, challengeMap) => {
-    if (!challengeName || !challengeMap) {
+  state => state.entities.block,
+  (challengeName, challengeMap, blockMap) => {
+    if (
+   !challengeName ||
+   !challengeMap ||
+   !blockMap ||
+   !challengeMap[challengeName] ||
+   !blockMap[challengeMap[challengeName].block]
+) {
       return {};
     }
     const challenge = challengeMap[challengeName];
     const challengeType = challenge && challenge.challengeType;
+    const challengeTitle = challengeTitleString({
+      blockName: blockMap[challenge.block].title || null,
+      title: challenge.title
+    });
     return {
       challenge,
+      challengeTitle,
```

Comment:
If we are using `challengeTitle` instead of `challenge.title` everywhere, then let's give it the respect it deserves and call it just `title`.